src/sdam/server.ts

```diff
@@ -378,17 +382,24 @@ export class Server extends TypedEventEmitter<ServerEvents> {
     if (!(error instanceof MongoError)) {
       return;
     }
-    if (error instanceof MongoNetworkError) {
-      if (!(error instanceof MongoNetworkTimeoutError) || isNetworkErrorBeforeHandshake(error)) {
-        // In load balanced mode we never mark the server as unknown and always
-        // clear for the specific service id.
-
-        if (!this.loadBalanced) {
-          error.addErrorLabel(MongoErrorLabel.ResetPool);
-          markServerUnknown(this, error);
-        } else if (connection) {
-          this.s.pool.clear(connection.serviceId);
-        }
+
+    // ignore stale errors
+    if (error.connectionGeneration && error.connectionGeneration < this.s.pool.generation) {
```

Comment:
```suggestion
    const isStaleGenerationError = error.connectionGeneration && error.connectionGeneration < this.s.pool.generation;
    if (isStaleGenerationError) {
```

comments can easily get out of date, but variable names cannot 🙂